drivers/adc/ad7768-1/ad77681.c

```diff
@@ -1295,6 +1298,103 @@ int32_t ad77681_programmable_filter(struct ad77681_dev *dev,
 	return ret;
 }
 
+int32_t ad77681_set_sinc3_dec_rate(struct ad77681_dev *dev, uint32_t dec_rate)
+{
+	uint32_t dec_rate_msb, dec_rate_lsb, max_dec_rate;
+	int ret;
+
+	max_dec_rate = dev->mclk / 100;
+	dec_rate = no_os_clamp_t(unsigned int, dec_rate, 32, max_dec_rate);
+	/*
+	 * Calculate the equivalent value to sinc3 decimation ratio
+	 * to be written on the SINC3_DECIMATION_RATE register:
+	 *  Value = (DEC_RATE / 32) -1
+	 */
+	dec_rate = NO_OS_DIV_ROUND_UP(dec_rate, 32) - 1;
+	dec_rate_msb = no_os_field_get(AD77681_SINC3_DEC_RATE_MSB_MSK, dec_rate);
+	dec_rate_lsb = no_os_field_get(AD77681_SINC3_DEC_RATE_LSB_MSK, dec_rate);
+
+	ret = ad77681_spi_reg_write(dev, AD77681_REG_SINC3_DEC_RATE_MSB, dec_rate_msb);
+	if (ret)
+		return ret;
+
+	ret = ad77681_spi_reg_write(dev, AD77681_REG_SINC3_DEC_RATE_LSB, dec_rate_lsb);
+	if (ret)
+		return ret;
+
+	dev->sinc3_osr = dec_rate;
+
+	return 0;
+}
+
+/**
+ * Compute the closest PGA gain index.
+ * @param dev The device structure.
+ * @param gain_int Integer part of desired gain.
+ * @param gain_fract Fractional part of desired gain.
+ * @param precision Fractional precision of gain_fract.
+ * @return Closest PGA gain index or negative error code.
+ */
+int32_t ad77681_calc_pga_gain(struct ad77681_dev *dev, int gain_int,
+			      int gain_fract, int precision)
+{
+	uint64_t gain_nano, tmp;
+	int gain_idx;
+
+	precision--;
+	gain_nano = gain_int * NANO + gain_fract;
+	if (gain_nano > ADAQ776X_GAIN_MAX_NANO)
+		return -EINVAL;
+
+	tmp = NO_OS_DIV_ROUND_CLOSEST_ULL(gain_nano << precision, NANO);
+	gain_nano = NO_OS_DIV_ROUND_CLOSEST_ULL(dev->vref, tmp);
+	if (dev->has_variable_aaf)
+		/* remove the AAF gain from the gain */
```

Comment:
i highly suggest you put brackets {} around the if branch even though functionally it's ok as is, that comment being there could be confusing for some and it might lead to bugs in the future if lines are added in the branch